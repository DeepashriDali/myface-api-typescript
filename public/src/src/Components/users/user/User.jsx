import React from "react";
import { Link } from "react-router-dom";
import "./User.scss"

export function User({ user }) {
    return <div className="user">
   
    <img className="profilePic" src={user.profileImageUrl} alt={user.name} /> 
    <div className="userInfo">
    <Link to = {`/users/${user.id}`}> <h2>{user.name}</h2> </Link>
    <p className="userName">{user.username}</p>
    <p className="email">{user.email} </p>
    </div>
    </div>
}