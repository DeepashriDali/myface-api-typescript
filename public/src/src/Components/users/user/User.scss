@import "../../../constants.scss";

.user{

    background-color: $darkerbackground;
    // background-color: $overlay;
    display:grid;
    grid-template-columns: 40vw 1fr ;
    grid-template-rows:  0.4fr 0.6fr;
    height: 40vw;
    margin:0.5rem;

    img {
    grid-column: 1/2;
    grid-row: 1/3;
    width:100%;
    margin:0;

}
    .profilePic{
        width: 6.5rem;
        height: 6.5rem;
        border: 0.25rem solid white;
        border-radius: 3.5rem;
 
    }

    h2{
    grid-column: 2/3;
    grid-row: 1/2;
    font-size: $mobile-subtitle;
    justify-self: center;
}
    p{
    grid-column: 2/3;
    grid-row: 2/3;
    font-size: $mobile-secondary;
    margin:0;
    // margin-left: 20vw;
    align-self: left;
    line-height: $mobile-subtitle-line-height;

}
p.userName{
    font-weight: bolder;
}

a:link { 
    text-decoration: none;
    color: $defaultbackground
}
a:visited { 
    text-decoration: none;
    color: $defaultbackground 
} 
a:hover { text-decoration: none;
    color: $defaultbackground 
}
a:active { text-decoration: none; 
    color: $defaultbackground}
}
.userDetailProfile {
    .profilePic {
        margin: 0.5rem;
        box-sizing: border-box;
        position: absolute;
        top:67vh;
        left:1vw;
        z-index: 200;
    }

    .userInfo {
        box-sizing: border-box;
        position: absolute;
        top:68vh;
        left:4vw;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: flex-start;
        margin-left: 10vw;
        padding-left: 4em;
        background-color: $overlay;
        height: 6.5rem;
        width: 100vw;
    }
}