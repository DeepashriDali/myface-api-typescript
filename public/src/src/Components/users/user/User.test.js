import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { User } from "./User";

test("renders message", () => {
  // Arrange
  const user = {
    name : "Sam",  
    username: "deeps",
    profileImageUrl: "Image url",
    email: "deepa@gmail.com",

  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>);
  const userElement = screen.getByText(/deeps/i);

  // Assert
  expect(userElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  // Arrange
  const user = {
    name : "Sam",  
    username: "deeps",
    profileImageUrl: "Image url",
    email: "deepa@gmail.com",

  };


  // Act
  const tree = renderer
    .create(<BrowserRouter><User user={user} /></BrowserRouter>)
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});