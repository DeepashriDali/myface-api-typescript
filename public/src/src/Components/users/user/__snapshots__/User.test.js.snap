// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="user undefined"
>
  <img
    alt="Sam"
    className="profilePic"
    src="Image url"
  />
  <div
    className="userInfo"
  >
    <a
      href="/users/undefined"
      onClick={[Function]}
    >
       
      <h2>
        Sam
      </h2>
       
    </a>
    <p
      className="userName"
    >
      deeps
    </p>
    <p
      className="email"
    >
      deepa@gmail.com
       
    </p>
  </div>
</div>
`;
